frame/system/src/lib.rs

```diff
@@ -1278,7 +1278,9 @@ impl<T: Config> Pallet<T> {
 	/// It is expected that light-clients could subscribe to this topics.
 	pub fn deposit_event_indexed(topics: &[T::Hash], event: T::Event) {
 		let block_number = Self::block_number();
-		// Don't populate events on genesis.
+		// Don't populate events on genesis for live chains.
+		// For tests, we don't check this to improve developer experience.
+		#[cfg(not(test))]
```

Comment:
I assume the aim to relief the pain of testing for crates that depend on `frame-system`. If so, then this won't work as intended because when `frame-system` is compiled not for running `frame-system`'s tests but for running tests in other crates the `#[cfg(test)]` will always NOT match. Thus this essentially a no-op for tests that live outside of the crate, meaning this if statement will still present in the final test binaries.